cli/azd/internal/errors.go

```diff
@@ -0,0 +1,49 @@
+package internal
+
+// ErrorWithSuggestion is a custom error type that includes a suggestion for the user
+type ErrorWithSuggestion struct {
+	Suggestion string
+	Err        error
+}
+
+// NewErrorWithSuggestion creates a new instance of the ErrorWithSuggestion
+func NewErrorWithSuggestion(err error, suggestion string) *ErrorWithSuggestion {
+	return &ErrorWithSuggestion{
+		Suggestion: suggestion,
+		Err:        err,
+	}
+}
```

Comment:
I'd opt to keep the constructor and make accessor functions for the elements that need to be exposed since the error should be immutable.  But you didn't like that pattern from previous PRs :)

Reply:
I'd definitely prefer the plain-old-struct that is clear on how it's used and laid out, but the constructor here is also fine.

> But you didn't like that pattern from previous PRs 🤷‍♂️

I understand that we all have differences in preferences. The main thing to think about is that while privacy and encapsulation has its benefits, adding protective layers that do not matter does result in bloat and complexity. This claim may seem egregious for this particular example, but when added up for the entirety of the codebase, it ends up adding a lot of extra overhead for maintainers in reading/writing code. A function/method necessarily signifies additional lines-of-code where things can potentially go wrong.